packages/request-client.js/src/http-metamask-data-access.ts

```diff
@@ -0,0 +1,306 @@
+import { Block } from '@requestnetwork/data-access';
+import { requestHashSubmitterArtifact } from '@requestnetwork/smart-contracts';
+import { DataAccessTypes } from '@requestnetwork/types';
+import Utils from '@requestnetwork/utils';
+import axios, { AxiosRequestConfig } from 'axios';
+import { ethers } from 'ethers';
+
+// Maximum number of retries to attempt when http requests to the Node fail
+const HTTP_REQUEST_MAX_RETRY = 3;
+
+// Delay between retry in ms
+const HTTP_REQUEST_RETRY_DELAY = 100;
+
+/**
+ * Exposes a Data-Access module over HTTP
+ */
+export default class HttpMetaMaskDataAccess implements DataAccessTypes.IDataAccess {
```

Comment:
I'm wondering if we can't ommit a good part of this class if we extend HttpDataAccess instead.